peps/pep-0757.rst

```diff
@@ -425,6 +408,28 @@ Python "native" layout.
 If later there are use cases for arbitrary layouts, new APIs can be
 added.
 
+Don't add :c:func:`PyLong_GetNativeLayout` function
+---------------------------------------------------
+
+Currently, all required information for :class:`int` import/export is
+already available via :c:func:`PyLong_GetInfo()` or
+:data:`sys.int_info`.  Native endianness of "digits" and current order
+of digits (least significant digit first) --- is a common denominator of
+all libraries for aribitrary precision integer arithmetic.
+
+The problem is that this API is for the stable ABI and not only for
+CPython, but for any Python implementation. For example, the JVM based
+GraalPy uses Java's ``BigDecimal`` for "big" integers, which has
+compatible internal representation: 4 bytes per digit, big-endian, most
+significant byte is in the zeroth element.
+
+People relying on CPython implementation details may assume CPython's
+:c:struct:`PyLongLayout` without even querying it and will just use the
+other APIs. Providing a convenient :c:func:`PyLong_GetNativeLayout`
+function is a way to reduce the risk of relying on CPython
+implementation details. By the way, the CPython native layout can also
+evolve in future Python versions.
+
```

Comment:
```suggestion
Currently, most required information for :class:`int` import/export is
already available via :c:func:`PyLong_GetInfo()` (and :data:`sys.int_info`).
We also can add more (like order of digits), this interface doesn't poses
any constraints on future evolution of the :type:`PyLongObject`.

The problem is that the :c:func:`PyLong_GetInfo()` returns a Python object,
:term:`named tuple`, not a convenient C structure and that might distract
people from using it in favor e.g. of current semi-private macros like :c:macro:`!PyLong_SHIFT` and :c:macro:`!PyLong_BASE`.


```

Reply:
I'm still not sure if we should rush with this change.  But if so, please take look on my edits.

I think the only arguments is: it's easy to use.  (BTW, it seems (I did GH search) the PyLong_GetInfo() isn't used outside of CPython.)